runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -509,12 +614,9 @@ function STHighlighter:reset()
   for client_id, state in pairs(self.client_state) do
     api.nvim_buf_clear_namespace(self.bufnr, state.namespace, 0, -1)
     state.current_result = {}
-    if state.active_request.request_id then
-      local client = vim.lsp.get_client_by_id(client_id)
-      assert(client)
-      client:cancel_request(state.active_request.request_id)
-      state.active_request = {}
-    end
+    local client = vim.lsp.get_client_by_id(client_id)
+    assert(client)
+    self:cancel_all_requests(client)
```

Comment:
Don't we need to also set `state.active_requests = {}` here?

On that note, we might want to do that inside `cancel_all_requests`.